pkg/login/social/social.go

```diff
@@ -79,7 +79,7 @@ func newSocialBase(name string, config *oauth2.Config, info *setting.OAuthInfo)
 	}
 }
 
-func NewOAuthService() {
+func NewOAuthService(cfg *setting.Cfg) {
```

Comment:
I see potential improvement of some of the other code here if you're up for it. Basically all references to reading settings using `settings.<setting>` could be replaced with using `cfg.<setting>` instead.

For example the line below `sec := setting.Raw.Section("auth." + name)` could be replaced with `sec := cfg.Raw.Section("auth." + name)`

Reply:
ok, will look into it